add stripe integration (#218)
val-town/val-town-docs: src/content/docs/integrations/stripe.mdx

---
title: Stripe
generated: 
description: | 

>     You can accept payments in Val Town using Stripe.

Reviewer: ```suggestion
    Accept payments in Val Town using Stripe
```

> lastUpdated: 2024-10-29

Reviewer: ```suggestion
```

---

import Val from "@components/Val.astro";
import gif from './stripe/stripe-try-it-out.gif'

> You can accept payments in Val Town. With this guide, you'll be able to accept payments in Val Town using Stripe.

Reviewer: ```suggestion
You can accept payments in Val Town with Stripe. This guide teaches you the simplest method: [Stripe Payment Links](https://stripe.com/payments/payment-links). Here's how it works:

1.  When users click to pay on your site, you redirect them to your stripe payment link
2. They pay with stripe
3. Stripe redirects them back to your site, where you can now store their payment information in a database, and thank them for their payment
```

> ## Obtain credentials from the Stripe

Reviewer: ```suggestion
## 1. Obtain credentials from the Stripe
```

Head over to the [Stripe dashboard](https://dashboard.stripe.com/dashboard) and get your secret key.

![stripe-keys.png](./stripe/stripe-keys.png)

> ## Add Stripe to your environment variables

Reviewer: ```suggestion
## 2. Add Stripe to your Val Town environment variables
```

Go to your [Val Town environment varibales](https://www.val.town/settings/environment-variables) and add your secret key to Val Town.

![stripe-env.png](./stripe/stripe-env.png)

## Ask Townie to generate our app

We can ask [Townie](https://www.val.town/townie), Val Town's bot, to generate our app for us. Here is an example prompt:

```
Create a tipping app with Stripe Checkout in the server. Use Tailwind CSS for 
styling and make the buttons purple. The preset tip options should 
be 1, 5, 10, 20.
```

![stripe-townie.png](./stripe/stripe-townie.png)

## Try it out!
You can now use Stripe in Val Town! Also feel free to [fork](https://www.val.town/v/vawogbemi/StripeDemohttps://www.val.town/v/vawogbemi/StripeDemo) the val below to try it out.

> <img src={gif.src} width={gif.width} />

Reviewer: This is great! I'd put this at the top, below my description of the steps, to excite users by how it looks all together
